### DISCUSSION TITLE
`rome format`: exit with error if diff

### DISCUSSION THREAD

**Discussion by elijaholmos** [rome/tools#3925] [Suggestions]
I'm playing around with Rome, particularly in a CI context. The `rome ci` command is cool, but I do enjoy having the ability to separately run the linter & formatter. My use case would be to split that up between two separate CI jobs for clarity.

What I'm running into is inconsistent output between `rome ci` and `rome format`. When running `rome ci` on my code, the following output is displayed:
```
$ C:\...\node_modules\.bin\rome ci ./graphql/scalars/FilterValue.ts
./graphql/scalars/FilterValue.ts format ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  × File content differs from formatting output

    13 │ ··return·value;
       │               +

Checked 1 files in 2ms
Error: errors where emitted while running checks
error Command failed with exit code 1.
```

However, `rome format` on the exact same file displays a different output:
```
$ C:\...\node_modules\.bin\rome format ./graphql/scalars/FilterValue.ts
./graphql/scalars/FilterValue.ts format ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  i Formatter would have printed the following content:

    13 │ ··return·value;
       │               +

Compared 1 files in 1708µs
Done in 0.24s.
```
I'm wondering why `ci` exits with an error when a diff is detected, but `format` doesn't. If this is verifiably intentional behavior, I would like to request an flag for `rome format` that instructs the command to exit with an error if formatting issues are detected.

**Comment by MichaReiser** [comment]
Hey @elijaholmos 

You can use `rome ci --linter-enabled=false <paths>` to only run the formatter and `--formatter=enabled=false` to only run the linter. 


Thank you for suggesting the exit code improvement. We're aware that the distinction between the two commands isn't entirely clear and that there are use cases that aren't served well at the moment. See #3443 for the ongoing discussion about the different commands.

**Reply by elijaholmos** [reply]
Thanks for the response @MichaReiser. I'll use the flags you recommended for now, as those satisfy my use case. Would still love to see consistency in the command exit codes going forward.
